### PR TITLE
Add @abstractstaticmethod and @abstractclassmethod

### PR DESCRIPTION
**Pull request by JakeNarkizian**
These combine the existing abstractmethod decorator with the classmethod, and staticmethod decorators respectively.

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
Commit title can be shortened to `Added decorators for abstract class/static methods`

Always capitalize the first word in a commit message and PR title.

**Comment by hannes-ucsc** [comment]
Sorry didn't mean to close.

**Comment by hannes-ucsc** [comment]
Commit message should be 

```
Added decorators for abstract class/static methods

These combine the existing @abstractmethod decorator with the @classmethod, 
and @staticmethod decorators respectively.
```

**Comment by hannes-ucsc** [comment]
No period at end of first line.

**Comment by hannes-ucsc** [comment]
The doctests fail for me. Do they work for you?

**Comment by JakeNarkizian** [comment]
My bad, I forgot to readd an import statement that I had taken out. It should be working now.

File: src/bd2k/util/objects.py [hunk 1]
```diff
@@ -2,6 +2,86 @@
...
+    http://stackoverflow.com/questions/11217878/python-2-7-combine-abc-abstractmethod-and-classmethod
+
+    >>> class DemoABC:
+    ...     from abc import ABCMeta
+    ...     __metaclass__ = ABCMeta
+    ...
+    ...     @abstractclassmethod
+    ...     def from_int(cls, n):
+    ...         return cls()
+    ...
```
Comment (hannes-ucsc) [inline src/bd2k/util/objects.py]: remove "..." but leave newline

File: src/bd2k/util/objects.py [hunk 2]
```diff
@@ -2,6 +2,86 @@
...
+    >>> class DemoConcrete(DemoABC):
+    ...     @classmethod
+    ...     def from_int(cls, n):
+    ...         return cls(2*n)
+    ...
+    ...     def __init__(self, n):
+    ...         print ('Initializing with %s' % n)
+    ...
+    >>> d = DemoConcrete(5)  # Succeeds by calling a concrete __init__()
```
Comment (hannes-ucsc) [inline src/bd2k/util/objects.py]: add empty line after

File: src/bd2k/util/objects.py [hunk 3]
```diff
@@ -2,6 +2,86 @@
...
+    Initializing with 5
+    >>> d = DemoConcrete.from_int(5)  # Succeeds by calling a concrete from_int()
```
Comment (hannes-ucsc) [inline src/bd2k/util/objects.py]: add empty line after

File: src/bd2k/util/objects.py [hunk 4]
```diff
@@ -2,6 +2,86 @@
...
+    Initializing with 10
+    >>> DemoABC()  # Fails because from_int() is abstract
```
Comment (hannes-ucsc) [inline src/bd2k/util/objects.py]: add empty line after

File: src/bd2k/util/objects.py [hunk 5]
```diff
@@ -2,6 +2,86 @@
...
+    """
+    This class defines a decorator that allows the decorated class to be both an abstract method
+    and a class method.
+
+    Shamelessly stolen from
+
+    http://stackoverflow.com/questions/11217878/python-2-7-combine-abc-abstractmethod-and-classmethod
+
+    >>> class DemoABC:
+    ...     from abc import ABCMeta
```
Comment (hannes-ucsc) [inline src/bd2k/util/objects.py]: wrong place for import

**Comment by hannes-ucsc** [comment]
src/bd2k/util/objects.py::bd2k.util.objects.abstractstaticmethod still failing

**Comment by hannes-ucsc** [comment]
And please remove unnecessary continuations from the doctests and add and empty line before every >>> statement.

**Comment by hannes-ucsc** [comment]
https://pypi.python.org/pypi?:action=display&name=bd2k-python-lib&version=1.13.dev14

**Pull request merged**